AK/PrintfImplementation.h

```diff
@@ -277,8 +278,8 @@ ALWAYS_INLINE int print_octal_number(PutChFunc putch, CharType*& bufptr, u64 num
     return field_width;
 }
 
-template<typename PutChFunc, typename CharType>
-ALWAYS_INLINE int print_string(PutChFunc putch, CharType*& bufptr, const char* str, size_t len, bool left_pad, size_t field_width, bool dot, size_t precision, bool has_fraction)
+template<typename PutChFunc, typename T, typename CharType>
+ALWAYS_INLINE int print_string(PutChFunc putch, CharType*& bufptr, T str, size_t len, bool left_pad, size_t field_width, bool dot, size_t precision, bool has_fraction)
```

Comment:
Dr.POSIX says the wide version has to be converted on the fly here:
https://pubs.opengroup.org/onlinepubs/9699919799/functions/printf.html
> If an l (ell) qualifier is present, the argument shall be a pointer to an array of type wchar_t. Wide characters from the array shall be converted to characters (each as if by a call to the [wcrtomb()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/wcrtomb.html) function, with the conversion state described by an mbstate_t object initialized to zero before the first wide character is converted) up to and including a terminating null wide character

Reply:
Yeah, I was also thinking about adding some tests :)

`wide version has to be converted on the fly here:` You mean that for every wide character to call [wcrtomb()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/wcrtomb.html) when calling printf? My patch wasn't meant to fix the printf with wide characters, but wprintf. [wprinf](https://pubs.opengroup.org/onlinepubs/9699919799/functions/wprintf.html) also contains requirements which are currently not implemented. My ambition wasn't to upgrade printf and wprinf to be in full compliance with the full POSIX, but solving one specific issue (wprintf not supporting wide characters at all). Not sure what to do now. I admit that the fix is not making everything 100% compliant with POSIX, but to make it compliant, it would be much bigger task.

Reply:
leaving a FIXME is fine too :)